pkg/agent/directive_handler_test.go

```diff
@@ -18,3 +19,33 @@ func TestHandleDirective(t *testing.T) {
 
 	dh.HandleDirective(ctx, a, d)
 }
+
+func TestHandleSecretSyncDirective(t *testing.T) {
+	// given
+	ctx := dlog.NewTestContext(t, false)
+
+	a := &agent.Agent{}
+	dh := &agent.BasicDirectiveHandler{}
+
+	d := &agentTypes.Directive{
+		ID: "one",
+		Commands: []*agentapi.Command{
+			{
+				SecretSyncCommand: &agentapi.SecretSyncCommand{
+					Name:      "my-secret",
+					Namespace: "my-namespace",
+					CommandId: "1234",
+					Action:    agentapi.SecretSyncCommand_SET,
+					Secret: map[string][]byte{
+						"my-key": []byte("abcd"),
+					},
+				},
+			},
+		},
+	}
+
+	// when
+	dh.HandleDirective(ctx, a, d)
+
+	// then
```

Comment:
I see this is copying a previous test that already exists but how are we asserting anything to be true? Is it just running this and then manually checking a cluster or ambassador cloud?

Reply:
Actually I thought I could actually test something, but the reality is that it is badly designed. I can't replace Agent for an interface or anything which would allow me to test its behavior.

It could for sure be improved, but I don't think that's the purpose of this PR.

Reply:
Ok, that is fine. I don't want to block this but I think that definitely is an area we should return to later to improve our test story here.